src/components/sidebarTableOfContents/index.tsx

```diff
@@ -183,7 +184,7 @@ export function SidebarTableOfContents() {
     }, observerOptions);
     const headings = tocItems.map(item => item.element);
     headings.forEach(heading => observer.observe(heading));
-    return () => headings.forEach(heading => observer.unobserve(heading));
+    return () => observer.disconnect();
```

Comment:
I think this is better, we can tear down the whole observer like this I believe? I can't even find `unobserve` here: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/disconnect 🤔